src/v/redpanda/application.cc

```diff
@@ -1345,6 +1345,30 @@ void application::start_redpanda(::stop_signal& app_signal) {
 
     storage.invoke_on_all(&storage::api::start).get();
 
+    ssx::background = _feature_table.invoke_on_all(
+      [this](features::feature_table& ft) -> ss::future<> {
+          try {
+              co_await ft.await_feature(features::feature::rpc_v2_by_default);
+              if (ss::this_shard_id() == 0) {
+                  vlog(_log.info, "Activating RPC protocol v2");
+              }
+              _connection_cache.local().set_default_transport_version(
+                rpc::transport_version::v2);
```

Comment:
> This still only activates it once controller log replay has advanced far
enough to update the feature table with the v2 feature: giving
feature table an early init from a kvstore snapshot will
be a separate change.

i'm not seeing how this is sufficient for what seems like some edge cases of the feature. just thinking through some scenarios here:

1. when upgrading _to_ 22.3 we'll have a mixed version cluster so `rpc_v2_by_default` won't turn on immediately. therefore, code that wants early connection v2 support either needs to (1) wait on 22.4 (23.1?) or (2) depend out-of-band on the feature to activate _and_ require a new connection or an upgraded connection?

2. with early controller log scan or kvstore reads for cached features, we'd still need a barrier to prevent rpc services from starting up and sending/receiving non-compliant early messages? this being in the background and running concurrent with other services starting up doesn't look sufficient for that case.

3. a new 22.3 node joining a cluster already running at 22.3 won't have a cached feature nor a controller log and will require rpc to discover this feature. does this place a special requirement on the feature rpc messages themselves? i guess this is sort of a special case of (1).